lib/edvs.js

```diff
@@ -0,0 +1,86 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const bedrock = require('bedrock');
+const brHttpsAgent = require('bedrock-https-agent');
+const {EdvClient} = require('edv-client');
+const keyResolver = require('./keyResolver');
+const kms = require('./kms');
+
+// load config defaults
+require('./config');
+const KMS_MODULE = 'ssm-v1';
+
+// module API
+const api = {};
+module.exports = api;
+
+/**
+ * Creates an EdvClient
+ *
+ *
+ * @return {Promise<EdvClient>} resolves to a EdvClient.
+ */
+api.create = async (
+  {invocationSigner, kmsClient, referenceId, profileId} = {}) => {
+  const edvBaseUrl = `${bedrock.config.server.baseUri}/edvs`;
+  const [keyAgreementKey, hmac] = await Promise.all([
+    kms.generateKey({
+      invocationSigner,
+      type: 'keyAgreement',
+      kmsClient,
+      kmsModule: KMS_MODULE
+    }),
+    kms.generateKey({
+      invocationSigner,
+      type: 'hmac',
+      kmsClient,
+      kmsModule: KMS_MODULE
+    })
+  ]);
+  // create edv
+  let config = {
+    sequence: 0,
+    controller: profileId,
+    keyAgreementKey: {id: keyAgreementKey.id, type: keyAgreementKey.type},
+    hmac: {id: hmac.id, type: hmac.type}
+  };
+  if(referenceId) {
+    config.referenceId = referenceId;
+  }
+  const {httpsAgent} = brHttpsAgent;
+  const capability = `${edvBaseUrl}/zcaps/configs`;
+  // TODO: Update to use the bedrock-edv-storage NodeJS API
+  config = await EdvClient.createEdv(
+    {url: edvBaseUrl, config, httpsAgent, invocationSigner, capability});
+  return new EdvClient(
+    {id: config.id, keyResolver, keyAgreementKey, hmac, httpsAgent});
+};
+
+api.get = async ({capabilityAgent, keystoreAgent, referenceId}) => {
+  const edvBaseUrl = `${bedrock.config.server.baseUri}/edvs`;
```

Comment:
Nothing actionable here, but I expect we're going to need to update this in the near future to support non-local EDVs.

Reply:
Added a TODO.